onnxruntime/contrib_ops/cpu/bert/embed_layer_norm.cc

```diff
@@ -87,7 +87,18 @@ Status EmbedLayerNorm<T>::Compute(OpKernelContext* context) const {
 
     int n = batch_size * sequence_length;
     concurrency::ThreadPool::TryBatchParallelFor(
+#if __cplusplus >= 202002L
+        context->GetOperatorThreadPool(), n, [=, this, &failed](ptrdiff_t index) {
```

Comment:
see this core guideline:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f54-when-writing-a-lambda-that-captures-this-or-any-class-data-member-dont-use--default-capture

I think they have a point. at least, we should consider how to make the code easier to understand.